components/webui/imports/ui/SearchView/datetime.js

```diff
@@ -0,0 +1,213 @@
+    return {
+        begin: dateTimeToDateWithoutChangingTimestamp(beginTime),
+        end: dateTimeToDateWithoutChangingTimestamp(endTime),
+    };
+};
+
+export const computePrevDayTimeRange = () => {
+    const endTime = DateTime.utc().minus({days: 1}).endOf("day");
+    const beginTime = endTime.startOf("day");
+    return {
+        begin: dateTimeToDateWithoutChangingTimestamp(beginTime),
+        end: dateTimeToDateWithoutChangingTimestamp(endTime),
+    };
+};
+
+export const computePrevWeekTimeRange = () => {
+    const endTime = DateTime.utc().minus({weeks: 1}).endOf("week");
+    const beginTime = endTime.startOf("week");
+    return {
+        begin: dateTimeToDateWithoutChangingTimestamp(beginTime),
+        end: dateTimeToDateWithoutChangingTimestamp(endTime),
+    };
+};
+
+export const computePrevMonthTimeRange = () => {
+    const endTime = DateTime.utc().minus({months: 1}).endOf("month");
+    const beginTime = endTime.startOf("month");
+    return {
+        begin: dateTimeToDateWithoutChangingTimestamp(beginTime),
+        end: dateTimeToDateWithoutChangingTimestamp(endTime),
+    };
+};
+
+export const computePrevYearTimeRange = () => {
+    const endTime = DateTime.utc().minus({years: 1}).endOf("year");
+    const beginTime = endTime.startOf("year");
+    return {
+        begin: dateTimeToDateWithoutChangingTimestamp(beginTime),
+        end: dateTimeToDateWithoutChangingTimestamp(endTime),
+    };
+};
+
+export const computeLast15MinTimeRange = () => {
+    const endTime = DateTime.utc();
+    const beginTime = endTime.minus({minutes: 15});
+    return {
+        begin: dateTimeToDateWithoutChangingTimestamp(beginTime),
+        end: dateTimeToDateWithoutChangingTimestamp(endTime),
+    };
+};
+
+export const computeLast60MinTimeRange = () => {
+    const endTime = DateTime.utc();
+    const beginTime = endTime.minus({minutes: 60});
+    return {
+        begin: dateTimeToDateWithoutChangingTimestamp(beginTime),
+        end: dateTimeToDateWithoutChangingTimestamp(endTime),
+    };
+};
+
+export const computeLast4HourTimeRange = () => {
+    const endTime = DateTime.utc();
+    const beginTime = endTime.minus({hours: 4});
+    return {
+        begin: dateTimeToDateWithoutChangingTimestamp(beginTime),
+        end: dateTimeToDateWithoutChangingTimestamp(endTime),
+    };
+};
+
+export const computeLast24HourTimeRange = () => {
+    const endTime = DateTime.utc();
+    const beginTime = endTime.minus({hours: 24});
+    return {
+        begin: dateTimeToDateWithoutChangingTimestamp(beginTime),
+        end: dateTimeToDateWithoutChangingTimestamp(endTime),
+    };
+};
+
+export const computeAllTimeRange = () => {
+    const endTime = DateTime.utc().plus({years: 1});
+    const beginTime = DateTime.fromMillis(0, {zone: "UTC"});
+    return {
+        begin: dateTimeToDateWithoutChangingTimestamp(beginTime),
+        end: dateTimeToDateWithoutChangingTimestamp(endTime),
+    };
+};
+
+export const cTimePresets = [
```

Comment:
* Typically we organize constants at the top of the file, before methods.
* I think we're following the convention of UPPER_SNAKE_CASE for constants elsewhere, right?

Reply:
* Did some refactoring to shorten the file from 214 lines to 110 lines with docs. 
* The table was placed after the function definitions because of the scope of the `const` arrow functions. We could place the table at the top by changing those function to named functions (lexical scoping, which I tried to avoid). However, it might not be a bad idea to place all  `export` to the end of files so that we could follow this structure:
  ```
  import ... ;
  ...
  export ... ;
  ```
  Feel free to raise objections.
* Updated the names to use UPPER_SNAKE_CASE.